mpacklog/python/sync_logger.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2022 Stéphane Caron
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import msgpack
+import queue
+
+from .serialize import serialize
+
+
+class SyncLogger:
+
+    """
+    Logger with synchronous I/O.
+
+    This logger exposes an API similar to AsyncLogger, but all I/O operations are synchronous.
+    """
+
+    def __init__(self, path):
+        self.__keep_going = True
+        self.path = path
+        self.queue = queue.Queue()
+
+        # Check if the file already exists, so that SyncLogger.write doesn't append to an existing file!
+        if os.path.exists(self.path):
+            raise FileExistsError(f"File {path} already exists!")
+
+    def put(self, message, write=False):
+        """
+        Puts a message in the queue.
+
+        @param message: The message to put in the queue.
```

Comment:
We use google style for docs in this project (rather than doxygen). I think `ruff` will complain on this one.

Reply:
Oh nevermind, we don't have `ruff` in the CI here. Let' s check it manually for now.